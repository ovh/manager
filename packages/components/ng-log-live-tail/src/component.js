import template from './template.html';
import controller from './controller';

export default {
  controller,
  template,
  transclude: {
    tile: '?tile',
    description: '?description',
  },
  bindings: {
    logApiUrl: '<',
    logKind: '<',
    logKeys: '<',
  },
};
