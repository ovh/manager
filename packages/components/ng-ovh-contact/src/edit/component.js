import controller from './controller';
import template from './template.html';

export default {
  require: {
    ovhContactCtrl: '^ovhContact',
  },
  template,
  controller,
  bindings: {
    options: '=?ovhContactChoiceOptions',
    customList: '=?ovhContactChoiceCustomList',
  },
};
