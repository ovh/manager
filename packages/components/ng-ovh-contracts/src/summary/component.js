import template from './template.html';

const component = {
  bindings: {
    contracts: '<items',
    inlineContracts: '<inlineItems',
    model: '=',
    name: '@?',
  },
  template,
};

export default component;
