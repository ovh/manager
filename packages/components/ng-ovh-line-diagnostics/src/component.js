import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    lineNumber: '@',
    serviceName: '@',
  },
  template,
  controller,
};
