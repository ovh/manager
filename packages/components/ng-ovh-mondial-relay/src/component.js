import controller from './controller';
import template from './template.html';

export default {
  bindings: {
    ngModel: '=?',
    userService: '=?',
    mondialRelayService: '=?',
  },
  template,
  controller,
};
