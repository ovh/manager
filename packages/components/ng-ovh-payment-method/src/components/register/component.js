import controller from './controller';
import template from './index.html';

export default {
  name: 'ovhPaymentMethodRegister',
  controller,
  template,
  bindings: {
    defaultPaymentMethodType: '@?',
    defaultPaymentMethodIntegration: '@?',
    model: '<',
    modelName: '@?',
    automaticSelect: '<?',
    automaticDefault: '<?',
    paymentMethodTypesOrder: '<?',
    paymentMethodTypesPerLine: '<?',
    registeredPaymentMethods: '<?',
    unregisteredPaymentMethods: '<?',
    excludedPaymentMethods: '<?',
    showSetAsDefaultChoice: '<?',
    forceSetAsDefaultChoice: '<?',
    onInitializationError: '&?',
    onInitialized: '&?',
    showDefaultExplanationTexts: '<?',
    defaultExplanationTextsOptions: '<?',
    required: '<?',
  },
  transclude: {
    introductionText: '?ovhPaymentMethodRegisterIntroductionText',
    explanationText: '?ovhPaymentMethodRegisterExplanationText',
    creditCardExplanationText:
      '?ovhPaymentMethodRegisterCreditCardExplanationText',
    sepaExplanationText: '?ovhPaymentMethodRegisterSepaExplanationText',
    paypalExplanationText: '?ovhPaymentMethodRegisterPaypalExplanationText',
    rupayExplanationText: '?ovhPaymentMethodRegisterRupayExplanationText',
  },
};
