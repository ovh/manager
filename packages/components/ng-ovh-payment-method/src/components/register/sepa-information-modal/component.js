import template from './template.html';

export default {
  template,
  bindings: {
    onConfirm: '&',
    isModalOpened: '<',
  },
};
