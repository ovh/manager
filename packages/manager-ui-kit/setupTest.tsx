import * as React from 'react';
import { ComponentType } from 'react';
import { I18nextProvider, initReactI18next } from 'react-i18next';
import i18n from 'i18next';
import {
  RenderOptions,
  RenderResult,
  render,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import 'element-internals-polyfill';
import { vi } from 'vitest';
import { QueryClient, QueryClientProvider } from '@tanstack/react-query';
import ActionMenuTransFR from './src/components/action-menu/translations/Messages_fr_FR.json';
import DatagridTransFR from './src/components/datagrid/translations/Messages_fr_FR.json';
import FiltersTransFR from './src/components/filters/translations/Messages_fr_FR.json';

i18n.use(initReactI18next).init({
  fallbackLng: 'fr_FR',
  interpolation: {
    escapeValue: false,
  },
  resources: {
    fr: {
      'action-menu': {
        ...ActionMenuTransFR,
      },
      datagrid: {
        ...DatagridTransFR,
      },
      filterAdd: {
        ...FiltersTransFR,
      },
    },
  },
});

export const queryClient = new QueryClient({
  defaultOptions: {
    mutations: {
      retry: false,
    },
    queries: {
      retry: false,
    },
  },
});

const Wrappers = ({ children }: { children: React.ReactElement }) => {
  return (
    <QueryClientProvider client={queryClient}>
      <I18nextProvider i18n={i18n}>{children}</I18nextProvider>
    </QueryClientProvider>
  );
};

const customRender = (
  ui: React.JSX.Element,
  options?: Omit<RenderOptions, 'queries'>,
): RenderResult =>
  render(ui, { wrapper: Wrappers as ComponentType, ...options });

export { customRender as render, cleanup };

// This polyfill exists because of an issue with jsdom and the EventTarget class
// when testing a component with an OdsDatepicker (addEventListener crashes at component initialization).
// Fix from issue: https://github.com/jsdom/jsdom/issues/2156
global.EventTarget = class {
  listeners = {};

  addEventListener(type, listener) {
    this.listeners = this.listeners || {};
    (this.listeners[type] || (this.listeners[type] = new Set())).add(listener);
  }

  removeEventListener(type, listener) {
    if (this.listeners && this.listeners[type]) {
      this.listeners[type].delete(listener);
    }
  }

  dispatchEvent(event) {
    this.listeners[event.type].forEach((listener) => listener(event));
    return !event.defaultPrevented;
  }
};

const ResizeObserverMock = vi.fn((callback) => {
  // Create a mock ResizeObserverEntry with the expected structure
  const mockEntry = {
    target: document.createElement('div'),
    contentRect: {
      width: 100,
      height: 100,
      top: 0,
      left: 0,
      bottom: 100,
      right: 100,
      x: 0,
      y: 0,
    },
    borderBoxSize: [{ width: 100, height: 100 }],
    contentBoxSize: [{ width: 100, height: 100 }],
    devicePixelContentBoxSize: [{ width: 100, height: 100 }],
  };

  // Call the callback with an array of entries as the real ResizeObserver does
  callback([mockEntry]);

  return {
    observe: vi.fn(),
    unobserve: vi.fn(),
    disconnect: vi.fn(),
  };
});

vi.stubGlobal('ResizeObserver', ResizeObserverMock);

// Mock getBoundingClientRect for virtualization and components
Element.prototype.getBoundingClientRect = vi.fn(() => ({
  width: 1000,
  height: 600,
  top: 0,
  left: 0,
  bottom: 600,
  right: 1000,
  x: 0,
  y: 0,
  toJSON: () => {},
}));

// Mock HTMLElement methods for components
HTMLElement.prototype.scrollTo = vi.fn();
HTMLElement.prototype.scrollIntoView = vi.fn();

// Mock scroll properties needed by ODS Modal
Object.defineProperty(Element.prototype, 'scrollLeft', {
  configurable: true,
  writable: true,
  value: 0,
});

Object.defineProperty(Element.prototype, 'scrollTop', {
  configurable: true,
  writable: true,
  value: 0,
});

Object.defineProperty(Element.prototype, 'scrollWidth', {
  configurable: true,
  value: 0,
});

Object.defineProperty(Element.prototype, 'scrollHeight', {
  configurable: true,
  value: 0,
});

// Suppress unhandled async errors from ODS Modal cleanup
const originalConsoleError = console.error;
console.error = (...args: any[]) => {
  const errorMessage = args[0]?.toString() || '';
  // Suppress specific ODS Modal/Zag.js cleanup errors
  if (
    errorMessage.includes("reading 'left'") ||
    errorMessage.includes('[@zag-js/dismissable]') ||
    errorMessage.includes('preventScroll') ||
    errorMessage.includes('Uncaught Exception')
  ) {
    return;
  }
  originalConsoleError(...args);
};

// Catch unhandled errors from ODS Modal async cleanup
if (typeof window !== 'undefined') {
  window.addEventListener('error', (event) => {
    const errorMessage = event.message || event.error?.message || '';
    // Suppress ODS Modal cleanup errors
    if (
      errorMessage.includes("reading 'left'") ||
      errorMessage.includes('[@zag-js/dismissable]')
    ) {
      event.preventDefault();
      event.stopPropagation();
      return false;
    }
    return true;
  });

  window.addEventListener('unhandledrejection', (event) => {
    const errorMessage = event.reason?.message || '';
    // Suppress ODS Modal cleanup errors
    if (
      errorMessage.includes("reading 'left'") ||
      errorMessage.includes('[@zag-js/dismissable]')
    ) {
      event.preventDefault();
      event.stopPropagation();
    }
  });
}
