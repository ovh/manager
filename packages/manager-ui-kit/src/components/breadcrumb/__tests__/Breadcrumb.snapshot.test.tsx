import { vitest } from 'vitest';
import { render } from '@/setupTest';
import { Breadcrumb } from '../Breadcrumb.component';

vitest.mock('../../../hooks/breadcrumb/useBreadcrumb', () => ({
  useBreadcrumb: vitest.fn(({ hideRootLabel }) => [
    { label: 'vRack services', href: '/', hideLabel: hideRootLabel },
  ]),
}));

describe('breadcrumb component snapshot', () => {
  it('should render 3 breadcrumb items when hideRootLabel is false', () => {
    const { asFragment } = render(
      <Breadcrumb
        rootLabel="vRack services"
        appName="vrack-services"
        hideRootLabel={false}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('should hide root label when hideRootLabel is true', () => {
    const { asFragment } = render(
      <Breadcrumb
        rootLabel="vRack services"
        appName="vrack-services"
        hideRootLabel={true}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
