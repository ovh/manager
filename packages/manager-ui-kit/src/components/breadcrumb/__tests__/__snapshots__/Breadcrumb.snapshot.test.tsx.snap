// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`breadcrumb component snapshot > should hide root label when hideRootLabel is true 1`] = `
<DocumentFragment>
  <nav
    class=""
    data-ods="breadcrumb"
    id=":r1:"
    role="navigation"
  >
    <ol
      class="_breadcrumb__list_1jn0o_2"
    />
  </nav>
</DocumentFragment>
`;

exports[`breadcrumb component snapshot > should render 3 breadcrumb items when hideRootLabel is false 1`] = `
<DocumentFragment>
  <nav
    class=""
    data-ods="breadcrumb"
    id=":r0:"
    role="navigation"
  >
    <ol
      class="_breadcrumb__list_1jn0o_2"
    >
      <li
        class="_breadcrumb-item_1bnix_2"
        data-ods="breadcrumb-item"
        role="listitem"
      >
        <span
          aria-current="page"
          data-ods="breadcrumb-link"
          href="/"
        >
          vRack services
        </span>
      </li>
    </ol>
  </nav>
</DocumentFragment>
`;
